src/tactic/simps.lean

```diff
@@ -543,14 +544,24 @@ derives two simp-lemmas:
   (this likely never happens).
 * When option `trace.simps.verbose` is true, `simps` will print the projections it finds and the
   lemmas it generates.
+* Use `@[to_additive, simps]` to apply both `to_additive` and `simps` to a definition, making sure
+  that `simps` comes after `to_additive`. This will also generate the additive versions of all
+  simp-lemmas. Note however, that the additive versions of the simp-lemmas always use the default
+  name generated by `to_additive`, even if a custom name is given for the additive version of the
+  definition.
   -/
 
 @[user_attribute] meta def simps_attr : user_attribute unit (list string × simps_cfg) :=
 { name := `simps,
   descr := "Automatically derive lemmas specifying the projections of this declaration.",
   parser := simps_parser,
   after_set := some $
-    λ n _ _, do (todo, cfg) ← simps_attr.get_param n, simps_tac n cfg todo }
+    λ n _ persistent, do
+      guard persistent <|> fail "`simps` currently cannot be used as a local attribute",
+      (todo, cfg) ← simps_attr.get_param n,
+      b ← has_attribute' `to_additive n,
+      let cfg' := if b then { attrs := cfg.attrs ++ [`to_additive], ..cfg } else cfg,
```

Comment:
Should this really be part of the `simps_attr` code? I'm thinking `simps_tac` could be called by other code generators, which probably also want to handle `to_additive` correctly.

Reply:
I agree, it seems relatively unlikely, but if it's not a big refactor Anne's suggestion seems cleaner.

Reply:
I was thinking that you might want to run the "bare" simps tactic, but that seems less likely than running the one that handles `to_additive` corectly. I changed it.